epochStart/notifier/epochStartSubscriptionHandler.go

```diff
@@ -61,7 +61,7 @@ func (essh *epochStartSubscriptionHandler) UnregisterHandler(handlerToUnregister
 
 // NotifyAll will call all the subscribed functions from the internal slice
 func (essh *epochStartSubscriptionHandler) NotifyAll(hdr data.HeaderHandler) {
-	essh.mutEpochStartHandler.RLock()
+	essh.mutEpochStartHandler.Lock()
```

Comment:
this is dangerous & should be tested thoroughly. Suggestion to call this sort whenever we add a new handler instead of each time we have a new block
Reason: we might get into deadlocks if the processing header will cause some flags or other changes that indirectly calls some trigger updates that call these functions.

Reply:
reverted the changes for now